bin/cron/dms_tasks

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env ruby
+
+# This script runs the AWS Data Migration Service tasks that replicate data from
+# our reporting RDS database to our Redshift database. It is highly tied to the
+# AWS DMS tasks in the us-east-1 region.
+# (https://console.aws.amazon.com/dms/home?region=us-east-1#tasks:)
+
+require 'cdo/hip_chat'
+require 'cdo/only_one'
+
+SECONDS_PER_MINUTE = 60
+REPLICATION_TASK_TYPE = 'reload-target'.freeze
+
+# @param task_execution_time_sec [Integer] The number of seconds the task's
+#   previous execution took.
+# @return [Integer] The number of seconds to wait before launching the next DMS
+#   task. This attempts to be a slight overestimate of the task's execution
+#   time.
+def get_delay(task_execution_time_sec)
+  # Note that we add 300 seconds since the time AWS DMS reports does not include
+  # the (non-negligible) amount of time spent "starting" the task.
+  if task_execution_time_sec > 60 * SECONDS_PER_MINUTE
+    return (task_execution_time_sec * 1.1 + 300).to_i
+  end
+  if task_execution_time_sec > 30 * SECONDS_PER_MINUTE
+    return (task_execution_time_sec * 1.2 + 300).to_i
+  end
+  return (task_execution_time_sec * 1.3 + 300).to_i
+end
+
+# The main method, responsible for creating a DMS client, fetching the set of
+# replication tasks, and starting the replication tasks.
+def main
+  dms_client = Aws::DatabaseMigrationService::Client.new(
+    region: 'us-east-1',
+    access_key_id: CDO.aws_access_key,
+    secret_access_key: CDO.aws_secret_key
```

Comment:
credentials don't need to be explicitly specified here, they will be automatically provided by the environment (either environment variables, shared config, or instance profile metadata)- see [documentation](http://docs.aws.amazon.com/sdkforruby/api/#Configuration) for reference. So just `Aws::DatabaseMigrationService::Client.new` should work